core/src/main/java/hudson/model/MyViewsProperty.java

```diff
@@ -139,7 +139,9 @@ public String getUrl() {
 
     @Override
     public void save() throws IOException {
-        user.save();
+        if (user != null) {
+            user.save();
```

Comment:
setUser is set via https://github.com/jenkinsci/jenkins/blob/e27b310065b3c036b5fc9d123f1d1d99d3058c00/core/src/main/java/hudson/model/User.java#L204

Which will be done later on not via jcasc

similar ish PR back in 2020 https://github.com/jenkinsci/jenkins/pull/4557